libecole/src/observation/strongbranchingscores.cpp

```diff
@@ -0,0 +1,118 @@
+#include <cmath>
+#include <cstddef>
+
+#include "ecole/observation/strongbranchingscores.hpp"
+#include "ecole/scip/type.hpp"
+
+#include <scip/scip.h>
+#include <scip/scipdefplugins.h>
+#include <scip/struct_branch.h>
+#include <scip/struct_lp.h>
+#include <scip/struct_scip.h>
+#include <scip/struct_var.h>
+#include <scip/utils.hpp>
+
+namespace ecole {
+namespace observation {
+
+StrongBranchingScores::StrongBranchingScores() {
+	pseudo_cands = true;
+}
+
+StrongBranchingScores::StrongBranchingScores(bool pseudo_candidates) {
+	pseudo_cands = pseudo_candidates;
+}
+
+auto StrongBranchingScores::obtain_observation(environment::State const& state)
+	-> nonstd::optional<StrongBranchingScoresObs> {
+
+	if (state.model.get_stage() == SCIP_STAGE_SOLVING) {
+
+		SCIP* scip = state.model.get_scip_ptr();
+
+		/* store original SCIP parameters */
+		SCIP_Bool integralcands;
+		SCIP_Bool scoreall;
+		SCIP_Bool collectscores;
+		SCIP_Bool donotbranch;
+		SCIP_Bool idempotent;
+
+		scip::call(
+			SCIPgetBoolParam,
+			scip,
+			"branching/vanillafullstrong/integralcands",
+			&integralcands);
+		scip::call(SCIPgetBoolParam, scip, "branching/vanillafullstrong/scoreall", &scoreall);
+		scip::call(
+			SCIPgetBoolParam,
+			scip,
+			"branching/vanillafullstrong/collectscores",
+			&collectscores);
+		scip::call(
+			SCIPgetBoolParam, scip, "branching/vanillafullstrong/donotbranch", &donotbranch);
+		scip::call(
+			SCIPgetBoolParam, scip, "branching/vanillafullstrong/idempotent", &idempotent);
+
+		/* set parameters for vanilla full strong branching  */
+		if (pseudo_cands) {
+			scip::call(
+				SCIPsetBoolParam, scip, "branching/vanillafullstrong/integralcands", true);
+		} else {
+			scip::call(
+				SCIPsetBoolParam, scip, "branching/vanillafullstrong/integralcands", false);
+		}
+		scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/scoreall", true);
+		scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/collectscores", true);
+		scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/donotbranch", true);
+		scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/idempotent", true);
```

Comment:
Have a look at 
https://github.com/ds4dm/ecole/blob/6c6b64f755bf0766ce739cda5688853bf0f02101/libecole/tests/src/scip/test-model.cpp#L73

You can reduce much verbosity here by using the syntax
```cpp
auto integralcands = state.model.get_param<bool>("branching/vanillafullstrong/integralcands");
```